stripe-ui-core/src/main/java/com/stripe/android/uicore/elements/TextFieldUI.kt

```diff
@@ -152,55 +157,58 @@ fun TextField(
 
     LaunchedEffect(fieldState) {
         // When field is in focus and full, move to next field so the user can keep typing
-        if (fieldState == TextFieldStateConstants.Valid.Full && hasFocus) {
+        if (fieldState == TextFieldStateConstants.Valid.Full && hasFocus.value) {
             focusManager.moveFocus(nextFocusDirection)
         }
     }
 
     val autofillReporter = LocalAutofillEventReporter.current
 
+    var selection by remember {
+        mutableStateOf<TextRange?>(null)
```

Comment:
Seems like this could be `TextRange(value.length)` as the initial value, no?

Reply:
Since `fieldValue` is a `Flow` and not a `StateFlow`, the initial value is always empty since we use `collectAsState("")` so the selection is not taken into account. Instead we wait for the first user interaction to set the selection. This can most likely be fixed after your PR updating everything to `StateFlow`.

Reply:
Oh right. Maybe if https://github.com/stripe/stripe-android/pull/8051 ever lands… 😉